elfloader-tool/include/types.h

```diff
@@ -6,28 +6,80 @@
 
 #pragma once
 
-typedef signed char int8_t;
-typedef short int16_t;
-typedef int int32_t;
-typedef long long int64_t;
+/* helper macro that ensure the passed macro gets evaluated first before the
+ * concatenation happens */
+#define _macro_make_string_helper(x)    # x
+#define _macro_make_string(x)           _macro_make_string_helper(x)
 
-typedef unsigned char uint8_t;
-typedef unsigned short uint16_t;
-typedef unsigned int uint32_t;
-typedef unsigned long long uint64_t;
+#define _macro_concat2_helper(x,y)      x ## y
+#define _macro_concat2(x,y)             _macro_concat2_helper(x,y)
+
+#define _macro_concat3_helper(x,y,z)    x ## y ## z
+#define _macro_concat3(x,y,z)           _macro_concat3_helper(x,y,z)
+
+
+typedef signed char     int8_t;
+typedef unsigned char   uint8_t;
+
+typedef signed short    int16_t;
+typedef unsigned short  uint16_t;
+
+typedef signed int      int32_t;
+typedef unsigned int    uint32_t;
+
+
+#if defined(__KERNEL_32__)
+
+#define _int64_type         long long
+#define _int64_type_fmt     ll  // for printf() formatting and integer suffix
+
+#elif defined(__KERNEL_64__)
+
+#define _int64_type         long long
```

Comment:
Can this not be a long here?

Reply:
That would have been my next question about this PR. So far I have just taken what was in the code before, which is `long long` for both 32-bit and 64-bit. Note that this is the same than what the kernels does. We can change this easily and just use `long` on 64-bit systems. **If nobody shouts, I will make this change**, as it is much more aligned with what our runtime libs do in user space. 
That leaves the question, if we should also try this change in the kernel then also. There there any reason (besides lazyness maybe) that is uses `long long` for u64 on 64-bit and not just `long`?

Reply:
I'm not sure why it was long long historically, if the ABI for the compiler is LP64 and not LLP64 then I agree we should be consistent (https://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models)